src/mode_RPiHQ_mean.cpp

```diff
@@ -217,18 +219,29 @@ void RPiHQcalcMean(const char* fileName, int asiExposure_us, double asiGain, ras
 	else {
 		currentRaspistillSetting.analoggain = newGain;
 	}
-	// min=1 us, max asiExposure
-	ExposureTime_s = std::min(asiExposure_us/US_IN_SEC, std::max(1 / US_IN_SEC, pow(2.0, double(currentModeMeanSetting.ExposureLevel)/pow(currentModeMeanSetting.shuttersteps,2.0)) / currentRaspistillSetting.analoggain));
+	// min=1 us, max=exposure_us
+	Log(5, "XXXX exposure_US/US_IN_SEC: %.2f, ExposureLevel:%d, shuttersteps: %.2f, analoggain: %.1f, pow=%.2f\n",
+		exposure_us/US_IN_SEC,
+		currentModeMeanSetting.ExposureLevel,
+		currentModeMeanSetting.shuttersteps,
+		currentRaspistillSetting.analoggain,
+		pow(2.0, double(currentModeMeanSetting.ExposureLevel)/pow(currentModeMeanSetting.shuttersteps,2.0)));
+
+	ExposureTime_s = std::min(exposure_us/US_IN_SEC, std::max(1 / US_IN_SEC, pow(2.0, double(currentModeMeanSetting.ExposureLevel)/pow(currentModeMeanSetting.shuttersteps,2.0)) / currentRaspistillSetting.analoggain));
 
 	//#############################################################################################################
 	// prepare for the next measurement
 	if (currentModeMeanSetting.quickstart > 0) {
+// xxxx  TODO: If already at the max exposure and we want to increase, then set quickstart to 0.
+// xxxx OR, if at a good exposure, set quickstart to 0.
 		currentModeMeanSetting.quickstart--;
 	}
 	// Exposure gilt fuer die naechste Messung
 	MeanCnt++;
 	exp_history[MeanCnt % currentModeMeanSetting.historySize] = currentModeMeanSetting.ExposureLevel;
 
 	currentRaspistillSetting.shutter_us = ExposureTime_s * US_IN_SEC;
-	printf("Mean: %1.4f (%1.4f) Exposure level:%d (%d) Exposure time:%1.8f analoggain:%1.2f\n", mean, mean_diff, currentModeMeanSetting.ExposureLevel, currentModeMeanSetting.ExposureLevel-exp_history[idx], ExposureTime_s, currentRaspistillSetting.analoggain);
+	Log(2, "  > Mean: %f, diff: %f, Exposure level:%d (%d), Exposure time:%1.8f ms, analoggain:%1.2f\n", mean, mean_diff, currentModeMeanSetting.ExposureLevel, currentModeMeanSetting.ExposureLevel-exp_history[idx], ExposureTime_s, currentRaspistillSetting.analoggain);
```

Comment:
wrong unit: use s, not ms
see: #824

Reply:
Fixed in commit 1f64e5c4380bb1b5f24bb4d9a5fcb8a9efe1fec5.